### PR TITLE
Use mapping file provider if available and bump AGP/Gradle

### LINKED ISSUE #6: getMappingFile is obsolete
A Gradle upgrade (5.6.4) gave me a new warning:

> WARNING: API 'variant.getMappingFile()' is obsolete and has been replaced with 'variant.getMappingFileProvider()'.
> It will be removed at the end of 2019.                                                                                                   
> For more information, see https://d.android.com/r/tools/task-configuration-avoidance. 

I ran gradle with `-Pandroid.debug.obsoleteApi=true` and it indicates that [it's coming from Sentry](https://github.com/getsentry/sentry-java/blob/v1.7.31_gradle_plugin/sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy#L236).

So, just making sure this is on your radar.

### PR DESCRIPTION
**Pull request by marandaneto**
Fix #6

### CODE REVIEW

File: src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy [hunk 1]
```diff
@@ -563,4 +563,17 @@ class SentryPlugin implements Plugin<Project> {
...
     }
+
+    /**
+     * Returns the mapping file
+     * @param variant the ApplicationVariant
+     * @return the file or null if not found
+     */
+    static File getMappingFile(ApplicationVariant variant) {
+        try {
+            return variant.getMappingFileProvider().get().singleFile
```
Comment (bruno-garcia) [inline src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: Grade new api of the hour?

**Comment by MBratkowski** [comment]
@bruno-garcia When can we expect merge this PR and release new version :)?

**Comment by bruno-garcia** [comment]
Don't recall why we didn't do that yet, I think just lack of time due to the work on release health  @marandaneto ?

**Comment by marandaneto** [comment]
> Don't recall why we didn't do that yet, I think just lack of time due to the work on release health @marandaneto ?

yep, lack of time.
I also have stated in the issue of this PR that it's not breaking anything (unless you use a non-stable version of Gradle/AGP).
https://github.com/getsentry/sentry-android-gradle-plugin/issues/6#issuecomment-597723710

I'll do it next week for sure though :)

File: src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy (hunk 1 above)
Comment (marandaneto) [reply src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: almost everything is a `Provider` now haha

**Comment by MBratkowski** [comment]
@marandaneto @bruno-garcia  Thanks a lot guys  :)

File: local.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Tue Feb 25 15:00:55 CET 2020
+sdk.dir=/Users/marandaneto/Library/Android/sdk
```
Comment (dzwicker) [inline local.properties]: This file should be ignored?

Comment (marandaneto) [reply local.properties]: @dzwicker it's outdated, when this repo has been moved, we lost the gitignore, but it already has been fixed.

**Pull request merged**